includes/class-freemius.php

```diff
@@ -3123,11 +3123,19 @@ private function is_matching_url( $sub_url, $url = '' ) {
                 return false;
             }
 
-            $url_params = array();
-            parse_str( parse_url( $url, PHP_URL_QUERY ), $url_params );
+            $url_components = parse_url( $url, PHP_URL_QUERY );
+            $url_params     = array();
 
-            $sub_url_params = array();
-            parse_str( parse_url( $sub_url, PHP_URL_QUERY ), $sub_url_params );
+            if ( ! empty( $url_components ) ) {

```

Comment:
@fajardoleo an immediate improvement we can do here is to extract the common logic in a function. Perhaps a dedicated `fs_parse_url_params` function in the `fs-core-functions`.

So it would become

```php
$url_params = fs_parse_url_params( $url );
$sub_url_params = fs_parge_url_params( $sub_url );
```